test/tests/interfaces/pbs_sched_interface_test.py

```diff
@@ -156,9 +182,129 @@ def test_set_and_unset_sched_attributes(self):
+
+    def test_set_default_sched_not_permitted(self):
+        """
+        Test setting partition on default scheduler
+        """
+        err_msg = "Operation is not permitted on default scheduler"
+        try:
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'partition': 'P1'},
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.assertTrue(err_msg in e.msg[0],
+                            "Error message is not expected")
+        try:
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'sched_priv': '/var/spool/somedir'},
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.assertTrue(err_msg in e.msg[0],
+                            "Error message is not expected")
+        try:
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'sched_log': '/var/spool/somedir'},
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.assertTrue(err_msg in e.msg[0],
+                            "Error message is not expected")
+
+    def test_sched_name_too_long(self):
+        """
+        Test creating a scheduler with name longer than 15 chars
+        """
+        try:
+            self.server.manager(MGR_CMD_CREATE, SCHED,
+                                runas=ROOT_USER, id="TestLongScheduler")
+        except PbsManagerError, e:
+            self.assertTrue("Scheduler name is too long" in e.msg[0],
+                            "Error message is not expected")
+
+    def test_set_default_sched_attrs(self):
+        """
+        Test setting scheduling and scheduler_iteration on default scheduler
+        and it updates server attributes and vice versa
+        """
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'False'},
+                            runas=ROOT_USER)
+        self.server.expect(SERVER, {'scheduling': 'False'})
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'scheduling': 'True'},
+                            runas=ROOT_USER)
+        self.server.manager(MGR_CMD_LIST, SCHED, id='default')
+        self.server.expect(SCHED, {'scheduling': 'True'},
+                           id='default', max_attempts=10)
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduler_iteration': 300},
+                            runas=ROOT_USER)
+        self.server.expect(SERVER, {'scheduler_iteration': 300})
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'scheduler_iteration': 500},
+                            runas=ROOT_USER)
+        self.server.manager(MGR_CMD_LIST, SCHED, id='default')
+        self.server.expect(SCHED, {'scheduler_iteration': 500},
+                           id='default', max_attempts=10)
+
```

Comment:
In the test suite the negative scenarios for the set/unset/create new attributes of multi-sched are missing.
Please add them.